apps/opik-backend/src/main/java/com/comet/opik/infrastructure/DatabaseUtils.java

```diff
@@ -25,4 +31,26 @@ public static DataSourceFactory filterProperties(DataSourceFactory dataSourceFac
 
         return dataSourceFactory;
     }
+
+    /**
+     * Handle SQL duplicate constraint violations in the state database (MySQL) with a specific error message.
+     *
+     * @param operation the database operation to execute
+     * @param errorMessage the error message to include in the exception if a duplicate constraint is violated
+     * @throws EntityAlreadyExistsException if a duplicate constraint violation occurs
+     * @throws UnableToExecuteStatementException if any other SQL exception occurs
+     */
+    public static void handleStateDbDuplicateConstraint(@NonNull Runnable operation,
```

Comment:
This already exists on EntityConstraintHandler

Reply:
Done in https://github.com/comet-ml/opik/pull/4019/commits/69eb2525d7b89628b47aec861b6bb8ee8b997cee